### PR TITLE
make sso backwards compatible

### PR DESCRIPTION
**Pull request by kubamahnert**

### CODE REVIEW

**Comment by panjan** [comment]
ok to merge

File: lib/gooddata/connection.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@ def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts
     # @param [String] login Email address used for logging into gooddata
     # @param [String] provider Name of SSO provider
     # @return [GoodData::Rest::Client] Instance of REST client
-    def connect_sso(login, provider, url, opts)
+    def connect_sso(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = {})
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has 4 parameters (LongParameterList)

File: lib/gooddata/connection.rb [hunk 2]
```diff
@@ -66,7 +66,7 @@ def with_connection(options = nil, second_options = nil, third_options = {}, &bl
     # @param [Hash] opts Additional options
     # @option opts [Fixnum] :validity Validity in seconds from 'now'
     # @return [String] URL which can be used for SSO logging in
-    def sso_url(login, provider, url = GoodData::Rest::Connection::DEFAULT_URL, opts = DEFAULT_SSO_OPTIONS)
+    def sso_url(login, provider, url, opts = DEFAULT_SSO_OPTIONS)
```
Comment (yenkins) [inline lib/gooddata/connection.rb]: Has approx 22 statements (TooManyStatements)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/922/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/772/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1040/ : SUCCESS in 2m 28s

**Pull request merged**